libs/remote_storage/src/lib.rs

```diff
@@ -164,6 +164,95 @@ impl GenericRemoteStorage {
             _ => None,
         }
     }
+
+    pub async fn upload_storage_object(
+        &self,
+        from: impl tokio::io::AsyncRead + Unpin + Send + Sync + 'static,
+        from_size_bytes: usize,
+        from_path: &Path,
+    ) -> anyhow::Result<()> {
+        async fn do_upload_storage_object<P, S>(
+            storage: &S,
+            from: impl tokio::io::AsyncRead + Unpin + Send + Sync + 'static,
+            from_size_bytes: usize,
+            from_path: &Path,
+        ) -> anyhow::Result<()>
+        where
+            P: std::fmt::Debug + Send + Sync + 'static,
+            S: RemoteStorage<RemoteObjectId = P> + Send + Sync + 'static,
+        {
+            let target_storage_path = storage.remote_object_id(from_path).with_context(|| {
+                format!(
+                    "Failed to get the storage path for source local path '{}'",
+                    from_path.display()
+                )
+            })?;
+
+            storage
+                .upload(from, from_size_bytes, &target_storage_path, None)
+                .await
+                .with_context(|| {
+                    format!(
+                        "Failed to upload from '{}' to storage path '{:?}'",
+                        from_path.display(),
+                        target_storage_path
+                    )
+                })
+        }
+
+        match self {
+            GenericRemoteStorage::Local(storage) => {
+                do_upload_storage_object(storage, from, from_size_bytes, from_path).await
+            }
+            GenericRemoteStorage::S3(storage) => {
+                do_upload_storage_object(storage, from, from_size_bytes, from_path).await
+            }
+        }
+    }
+
+    pub async fn download_storage_object(
+        &self,
+        byte_range: Option<(u64, Option<u64>)>,
```

Comment:
All the callers pass `byte_range = None`. How about just removing it?

Reply:
Safekeeper uses an offset to download things, so one place uses this param.